pallets/migration/src/weights.rs

```diff
@@ -0,0 +1,33 @@
+//! Autogenerated weights for pallet_fees
+//!
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 3.0.0
+//! DATE: 2021-06-07, STEPS: `[50, ]`, REPEAT: 100, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! EXECUTION: Some(Wasm), WASM-EXECUTION: Compiled, CHAIN: Some("charcoal-local"), DB CACHE: 128
+
+// Executed Command:
+// target/release/centrifuge-chain
+// benchmark
+// --chain=charcoal-local
+// --steps=50
+// --repeat=100
+// --pallet=pallet_fees
+// --extrinsic=*
+// --execution=wasm
+// --wasm-execution=compiled
+// --heap-pages=4096
+// --output=./pallets/fees/src/weights.rs
+// --template=./scripts/frame-weight-template.hbs
+
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::{
+	traits::Get,
+	weights::{constants::RocksDbWeight, Weight},
+};
+use sp_std::marker::PhantomData;
+
+/// Weight functions needed for pallet_fees.
+pub trait WeightInfo {
+	fn upgrade_system_account() -> Weight;
```

Comment:
I like this more than using `on_runtime_upgrade` for migration initialisation. Then we dont have to start immediately but rather wait until any other core migrations happen

Reply:
Ah. Okay. Yes, we could do this with sudo of course. My idea was to control all logic from the `MigrationPallet`. Somethink like:

```pseudocode
SOME_PALLET::on_runtime_upgrade() {
    if pallet_migration_manager::<UpgradeTriggered<T>> {
          // Execute actual logic here and return weight
    } else {
         Zero::zero()
    }
```

With this, a runtime upgrade would solely set the trigger in the migration manager which itself then blocks the next `n` blocks via `on_initialize` and calls the respective functions (including pallet specific `on_runtime_upgrade`s as we need them to access the private methods of some pallets.

But again. Your approach seems even smoother maybe. As we can even trigger some migrations without an official runtime upgrade. We could maybe migrate the pallets first. And once this is through, we trigger the data migration. The more I think about it, the more I like it. 

I wanted the migration manager to be something, that we can use generically everytime we need a `multi-block` upgrades, hence my idea of controlling the pallets `on_runtime_upgrades`. But this might not be necessary.